cmake/BuildOptions.cmake

```diff
@@ -6,6 +6,19 @@ enum_option(PLATFORM "Desktop;Web;Android;Raspberry Pi;DRM" "Platform to build f
 
 enum_option(OPENGL_VERSION "OFF;3.3;2.1;1.1;ES 2.0" "Force a specific OpenGL Version?")
 
+# This helps support the case where emsdk toolchain file is used
+# either by setting it with -DCMAKE_TOOLCHAIN_FILE=<path_to_emsdk>/upstream/emscripten/cmake/Modules/Platform/Emscripten.cmake
+# or by using "emcmake cmake -B build -S ." as described in https://emscripten.org/docs/compiling/Building-Projects.html
+if(EMSCRIPTEN)
+    SET(PLATFORM Web CACHE STRING "Forcing PLATFORM_WEB because EMSCRIPTEN was detected")
+endif()
```

Comment:
1. Most CMake standards I've used have used `CMakeOptions.txt` for this, not `BuildOptions.cmake`... I'm okay with this change, just seems off-standard.
2. You're `SET`ing in an options file. Intentional?

Reply:
@RobLoach I will return the standard CMakeOptions.txt then to be more inline with other standards then. I though the set belongs nicely there but I guess it doesn't. It forces the PLATFORM option to default to WEB when compiling with Emscripten which is a web compiler.